ndc-client/src/models.rs

```diff
@@ -350,13 +350,46 @@ pub enum Aggregate {
 }
 // ANCHOR_END: Aggregate
 
+// ANCHOR: NestedObject
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[schemars(title = "NestedObject")]
+pub struct NestedObject {
+    pub fields: IndexMap<String, Field>,
+}
+// ANCHOR_END: NestedObject
+
+// ANCHOR: NestedArray
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[schemars(title = "NestedArray")]
+pub struct NestedArray {
+    pub fields: Box<Option<NestedField>>,
```

Comment:
What does `None` represent here?

Reply:
`None` means that there are no nested selections for the array elements, e.g. if the elements are scalars.

Reply:
Example:
```json
    "query": {
        "fields": {
            "departments": {
                "type":"column",
                "column": "departments",
                "fields": {
                    "type": "array"
                }
            }
        }
```
Field `departments` is an array of strings so there are no subselections within the array elements

Reply:
But then why not use `Field::Column { fields: None, ... }`?

Reply:
We _could_ do that now, but in the future we might want to perform other operations on arrays, e.g. sorting or filtering.

Reply:
But those could be added as new constructors to `NestedField`, I think? If we're concerned about that, then we arguably should make `NestedArray` an enum now.